src/lib/tecs/TECS.hpp

```diff
@@ -49,351 +49,616 @@
-	float _STE_error{0.0f};						///< specific total energy error (m**2/sec**2)
-	float _STE_rate_error{0.0f};					///< specific total energy rate error (m**2/sec**3)
-	float _SEB_error{0.0f};						///< specific energy balance error (m**2/sec**2)
-	float _SEB_rate_error{0.0f};					///< specific energy balance rate error (m**2/sec**3)
-
-	// speed height weighting
-	float _SPE_weighting{1.0f};
-	float _SKE_weighting{1.0f};
-
-	// time steps (non-fixed)
-	float _dt{DT_DEFAULT};						///< Time since last update of main TECS loop (sec)
-	static constexpr float DT_DEFAULT = 0.02f;			///< default value for _dt (sec)
+	float _equivalent_airspeed_trim{15.0f};				///< equivalent cruise airspeed for airspeed less mode (m/sec)
 
 	// controller mode logic
-	float _percent_undersped{0.0f};					///< a continuous representation of how "undersped" the TAS is [0,1]
-	bool _detect_underspeed_enabled{true};				///< true when underspeed detection is enabled
 	bool _uncommanded_descent_recovery{false};			///< true when a continuous descent caused by an unachievable airspeed demand has been detected
-	bool _climbout_mode_active{false};				///< true when in climbout mode
 	bool _airspeed_enabled{false};					///< true when airspeed use has been enabled
-	bool _states_initialized{false};				///< true when TECS states have been iniitalized
+	bool _detect_underspeed_enabled{false};				///< true when underspeed detection is enabled
 
-	/**
-	 * Update the airspeed internal state using a second order complementary filter
-	 */
-	void _update_speed_states(float airspeed_setpoint, float equivalent_airspeed, float cas_to_tas);
-
-	/**
-	 * Update the desired airspeed
-	 */
-	void _update_speed_setpoint();
+	static constexpr float DT_MIN = 0.001f;				///< minimum allowed value of _dt (sec)
+	static constexpr float DT_MAX = 1.0f;				///< max value of _dt allowed before a filter state reset is performed (sec)
 
-	/**
-	 * Calculate desired height rate from altitude demand
-	 */
-	void runAltitudeControllerSmoothVelocity(float alt_sp_amsl_m, float target_climbrate_m_s, float target_sinkrate_m_s,
-			float alt_amsl);
+	static constexpr float _jerk_max = 1000.0f;			///< Magnitude of the maximum jerk allowed [m/s³].
+	static constexpr float _tas_error_percentage =
+		0.15f;		///< [0,1] percentage of true airspeed trim corresponding to expected (safe) true airspeed tracking errors
 
-	/**
-	 * Detect how undersped the aircraft is
-	 */
-	void _detect_underspeed();
+	DebugOutput _debug_status{};
 
-	/**
-	 * Update specific energy
-	 */
-	void _update_energy_estimates();
+	// Params
+	/// Airspeed filter parameters.
+	TECSAirspeedFilter::Param _airspeed_param{
+		.equivalent_airspeed_trim = 0.0f,
+		.airspeed_measurement_std_dev = 0.2f,
+		.airspeed_rate_measurement_std_dev = 0.05f,
+		.airspeed_rate_noise_std_dev = 0.02
+	};
+	/// Reference model parameters.
+	TECSReferenceModel::Param _reference_param{
+		.target_climbrate = 2.0f,
+		.target_sinkrate = 2.0f,
+		.jerk_max = _jerk_max,
```

Comment:
```suggestion
		.jerk_max = 1000.f,
```
What is the motivation to keep this constant definition separated?

Reply:
In the older implementation, the value was used moret han once. Can know be removed since it is only used at one place.